lib/nexussw/lxd/transport/mixins/helpers/folder_txfr.rb

```diff
@@ -1,41 +1,75 @@
+require 'zlib'
+require 'archive/tar/minitar'
+
 module NexusSW
   module LXD
     class Transport
       module Mixins
         module Helpers
-          module UploadFolder
+          module FolderTxfr
             def upload_folder(local_path, path, options = {})
               upload_using_tarball(local_path, path, options) || upload_files_individually(local_path, path, options)
             end
 
+            def download_folder(path, local_path)
+              download_using_tarball(path, local_path) || download_files_individually(path, local_path)
+            end
+
             def upload_files_individually(local_path, path, options = {})
+              dest = File.join(path, File.basename(local_path))
+              execute 'mkdir -p ' + dest, capture: false # for parity with tarball extract
```

Comment:
add .error!

and probably remove capture: false until the rest api race is handled